### PR TITLE
fixed irrxml dependency problem in Codec_Assimp

### LINKED ISSUE #2072: when building with assimp, error occurs for libIrrXML.a
[ 45%] Linking CXX shared library ../../lib/Plugin_PCZSceneManager.so
[ 45%] Built target Plugin_PCZSceneManager
make[2]: *** No rule to make target 'Dependencies/lib/libIrrXML.a', needed by 'lib/Codec_Assimp.so.1.12.12'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 45%] Building CXX object PlugIns/Assimp/CMakeFiles/Codec_Assimp.dir/src/AssimpLoader.cpp.o
[ 45%] Building CXX object Components/Paging/CMakeFiles/OgrePaging.dir/src/OgrePageContent.cpp.o
[ 46%] Building CXX object Components/Paging/CMakeFiles/OgrePaging.dir/src/OgrePageContentCollection.cpp.o
[ 46%] Building CXX object Components/Paging/CMakeFiles/OgrePaging.dir/src/OgrePageManager.cpp.o
[ 46%] Building CXX object Components/Paging/CMakeFiles/OgrePaging.dir/src/OgrePagedWorld.cpp.o
make[1]: *** [CMakeFiles/Makefile2:1141: PlugIns/Assimp/CMakeFiles/Codec_Assimp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 46%] Building CXX object Components/Paging/CMakeFiles/OgrePaging.dir/src/OgrePagedWorldSection.cpp.o
[ 46%] Building CXX object Components/Paging/CMakeFiles/OgrePaging.dir/src/OgreSimplePageContentCollection.cpp.o
[ 46%] Linking CXX shared library ../../lib/libOgrePaging.so
[ 46%] Built target OgrePaging
make: *** [Makefile:156: all] Error 2

I use the following commands to build on manjaro linux : 
`cmake .. -DCMAKE_BUILD_TYPE=Release && make -j2`

Please help

### PR DESCRIPTION
**Pull request by brightprogrammer**
Fixed Codec_Assimp dependency problem with IrrXML. 

fixes  #2072

### CODE REVIEW

File: PlugIns/GLSLang/src/OgreGLSLang.cpp [hunk 1]
```diff
@@ -109,19 +109,19 @@ const TBuiltInResource DefaultTBuiltInResource = {
...
-        /* .doWhileLoops = */ 1,
-        /* .generalUniformIndexing = */ 1,
-        /* .generalAttributeMatrixVectorIndexing = */ 1,
-        /* .generalVaryingIndexing = */ 1,
-        /* .generalSamplerIndexing = */ 1,
-        /* .generalVariableIndexing = */ 1,
-        /* .generalConstantMatrixVectorIndexing = */ 1,
-    }};
-}
+        .limits = {
```
Comment (paroj) [inline PlugIns/GLSLang/src/OgreGLSLang.cpp:112]: unrelated changes

**Comment by paroj** [comment]
irrXML should be have been built by assimp:
https://github.com/assimp/assimp/tree/v5.0.1/contrib/irrXML

at least you dont have to download the sources separately.

Also, note that Assimp master does not use irrXML any more.

**Comment by brightprogrammer** [comment]
Well, it gave an error on build which I mentioned in an issue and it was marked as a known issue, meaning it wasn't resolved, but now the build completes successfully, so I guess it can be acceptable?

**Comment by paroj** [comment]
the Linux build breaks with these changes

**Comment by brightprogrammer** [comment]
I can't figure out why that happens. It builds on my system and I use manjaro.

File: CMake/Dependencies.cmake [hunk 2]
```diff
@@ -82,6 +82,15 @@ if(OGRE_BUILD_DEPENDENCIES AND NOT EXISTS ${OGREDEPS_PATH})
...
         --build ${PROJECT_BINARY_DIR}/pugixml-1.10 ${BUILD_COMMAND_OPTS})
 
+    
+    # download and extract irrxml so that we can build it later
+    message(STATUS "Building IrrXML")
+    # download
+    file(DOWNLOAD https://nchc.dl.sourceforge.net/project/irrlicht/irrXML%20SDK/1.2/irrxml-1.2.zip ${PROJECT_BINARY_DIR}/irrxml-1.2.zip)
+    # extract
+    execute_process(COMMAND ${CMAKE_COMMAND}
+    -E tar xf irrxml-1.2.zip WORKING_DIRECTORY ${PROJECT_BINARY_DIR})
```
Comment (Joilnen) [inline CMake/Dependencies.cmake:92]: tar is not able to decompress zip files

**Comment by paroj** [comment]
no progress for 2 months
